dev-scheme/guile/guile-2.2.3.recipe

```diff
@@ -0,0 +1,118 @@
+applications. Using Guile in an application allows programmers to write \
+plug-ins, or modules (there are many names, but the concept is essentially \
+the same) and users to use them to have an application fit their needs."
+HOMEPAGE="https://www.gnu.org/software/guile/"
+COPYRIGHT="1993-2018 Aubrey Jaffer, George Carrette, et al."
+LICENSE="GNU LGPL v3"
+REVISION="1"
+SOURCE_URI="https://ftpmirror.gnu.org/guile/guile-$portVersion.tar.gz
+	https://ftp.gnu.org/gnu/guile/guile-$portVersion.tar.gz"
+CHECKSUM_SHA256="87ee07caef33c97ddc74bf3c29ce7628cfac12061f573e4a29a3a1176754610a"
+PATCHES="guile-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+commandSuffix=$secondaryArchSuffix
+commandBinDir=$binDir
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	commandSuffix=
+	commandBinDir=$prefix/bin
+fi
+
+libguileVersion="1.3.0"
+libguileVersionCompat="$libguileVersion compat >= ${libguileVersion%%.*}"
+portVersionCompat="$portVersion compat >= ${portVersion%%.*}"
+portVers="${portVersion%.*}"
+
+PROVIDES="
+	guile$secondaryArchSuffix = $portVersionCompat
+	cmd:guild$commandSuffix = $portVersion
+	cmd:guile$commandSuffix = $portVersion
+	cmd:guile_snarf$commandSuffix = $portVersion
+	cmd:guile_tools$commandSuffix = $portVersion
+	lib:libguile_$portVers$secondaryArchSuffix = $libguileVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libffi$secondaryArchSuffix
+	lib:libgc$secondaryArchSuffix
+	lib:libgmp$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
+	lib:libltdl$secondaryArchSuffix
+	lib:libncurses$secondaryArchSuffix
+	lib:libreadline$secondaryArchSuffix
+	lib:libunistring$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	guile${secondaryArchSuffix}_devel = $portVersionCompat
+	cmd:guile_config$commandSuffix = $portVersion
+	devel:libguile_$portVers$secondaryArchSuffix = $libguileVersionCompat
+	"
+REQUIRES_devel="
+	guile$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libffi$secondaryArchSuffix
+	devel:libgc$secondaryArchSuffix
+	devel:libgmp$secondaryArchSuffix
+	devel:libiconv$secondaryArchSuffix
+	devel:libintl$secondaryArchSuffix
+	devel:libltdl$secondaryArchSuffix
+	devel:libncurses$secondaryArchSuffix
+	devel:libreadline$secondaryArchSuffix
+	devel:libunistring$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:awk
+	cmd:gcc$secondaryArchSuffix
+	cmd:gperf
+	cmd:grep
+	cmd:make
+	cmd:ld
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:sed
+	"
+
+defineDebugInfoPackage guile$secondaryArchSuffix \
+	"$libDir"/libguile-$portVers.so.$libguileVersion
+
+BUILD()
+{
+	CPPFLAGS="-D_BSD_SOURCE" LIBS="-lbsd -lnetwork" \
+	runConfigure --omit-dirs binDir ./configure \
+		--with-threads
+
+	# No MADV_DONTNEED, yet
+	sed --in-place '/HAVE_SYS_MMAN_H/d' config.h
+
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+
+	rm -f "$libDir"/lib*.la
+	rm -f "$libDir"/guile/$portVers/extensions/guile-readline.la
```

Comment:
```
package list -p guile_x86-2.2.3-1-x86_gcc2.hpkg | grep /extensions/
lib/x86/guile/2.2/extensions/guile-readline.a
lib/x86/guile/2.2/extensions/guile-readline.so
lib/x86/guile/2.2/extensions/guile-readline.so.0
lib/x86/guile/2.2/extensions/guile-readline.so.0.0.0
```
We can't move `guile-readline.a` to the devel package with `prepareInstalledDevelLib` because it would not move it to `$developLibDir/guile/2.2/extensions/` but to `$developLibDir/`.
(It seems it's only appropriate to call `prepareInstalledDevelLib` for libs that are in `$libDir`, not for libs which are in a subdirectory.)
So, instead of dropping just the libtool file, shall we also drop the static lib?
```diff
-	rm -f "$libDir"/guile/$portVers/extensions/guile-readline.la
+	rm -f "$libDir"/guile/$portVers/extensions/guile-readline.*a
```
(This is not the main lib, `libguile-2.2.a` which `prepareInstalledDevelLib` already moves to the devel package.)

Reply:
I believe using `find` would be much simpler in this case. It would also handle the case where more extensions are added.

```diff
-	rm -f "$libDir"/lib*.la
-	rm -f "$libDir"/guile/$portVers/extensions/guile-readline.la
+	find -name '*.la' -type f -delete

Reply:
Hmm, there are only 2 libtool files, so I think it is nicer without the find command.

BTW, I'm suggesting one `rm` to delete the libtool file of the main lib (`libguile-2.2.la`), and another `rm` to delete both the libtool file of the extension lib (`guile-readline.la`) as well as the static lib itself, `guile-readline.a`:
```bash
	rm -f "$libDir"/lib*.la
	rm -f "$libDir"/guile/$portVers/extensions/guile-readline.*a
```

Reply:
It's nice until they add more, but I guess that could be for later when they actually do so.

About the static lib, I don't know how guile work, but maybe the static lib is necessary some how?

Reply:
The extension available in our build of guile is currently present as a static lib, `guile-readline.a` as well as a shared lib, `guile-readline.so.0.0.0`.
You are probably right that guile might need that extension's static lib when guile is told to compile some program. But since it also comes with the shared lib, I believe guile can be told to create a compiled runtime that dynamically links to the shared lib.

I therefore see 3 scenari:
1) we keep that static lib in the base package. That's OK for me, although I think it would be better in the devel package.
2) we move the static lib to the devel package, but in that case we would have to create the symbolic links just like `prepareInstalledDevelLib` does when it is given the (base)name of a shared lib in `$libDir`. We could try this, but we won't be sure it is OK until we have used guile to compile something that needs the `guile-readline.a` static lib. I'm OK with this option, but it will require some investigation.
3) we choose to drop the static lib. I think this is the best scenario because it is easy to "implement" and is also easy to revert, in case we find out that it's not possible to compile a program against `guile-readline.so.0` if the corresponding static lib is missing. But this would be weird.

So I belive the best scenari are 3 and 1. The PR is currently doing scenario 1. Shall we try scenario 3?

To address your comment about future extensions, shall we use the following commands?
```bash
	rm -f "$libDir"/lib*.la
	rm -f "$libDir"/guile/$portVers/extensions/*.la
	rm -f "$libDir"/guile/$portVers/extensions/*.a
```
(Of course, the last command should be dropped if we stay in scenario 1.)